torch/ao/nn/quantized/dynamic/modules/rnn.py

```diff
@@ -61,7 +62,7 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
 
 class RNNBase(torch.nn.Module):
 
-    _FLOAT_MODULE = nn.RNNBase
+    _FLOAT_MODULE: Type[nn.RNNBase] = nn.RNNBase            # Annotation to ensure subclass attributes check against known type
```

Comment:
Could you move the comment to the line above please 
```suggestion
     # Annotation to ensure subclass attributes check against known type
    _FLOAT_MODULE: Type[nn.RNNBase] = nn.RNNBase
```

Separately, to aid my understanding could you please give a minimal example of when this type annotation is helpful?

Reply:
Thank you for all of your help! 

Locally I would receive a mypy error without the type annotation: `torch/ao/nn/quantized/dynamic/modules/rnn.py:643:21: error: Incompatible types in assignment (expression has type "Type[GRU]", base class "RNNBase" defined the type as "RNNBase")  [assignment]`. Adding the `Type[nn.RNNBase]` type annotation resolves this issue. I'm looking into this now.

Reply:
It seems as though the ` _FLOAT_MODULE: Type[nn.RNNBase] = nn.RNNBase` is required to pass the `mypy` checks. The type annotations for GRU and LSTM classes for stylistic consistency. 

It seems as though the additional specifications in the `rnn.py` triggered the `Incompatible types in assignment` errors, but I'm not entirely sure why...

Reply:
This is no longer an issue given the changes to `rnn.py`. I removed the changes to this file.